File: src/hello_world_py_pkg/hello_world_py_pkg/hello_world_node.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

def main(args=None):
    rclpy.init(args=args)
    # All code goes between rclpy.init() and rclpy.shutdown()

    node = Node('new_test_node')
    node.get_logger().info('Hello, world!')

    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/hello_world_cpp_pkg/src/hello_world_node.cpp
#include "rclcpp/rclcpp.hpp"

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<rclcpp::Node>("hello_world_node");
    RCLCPP_INFO(node->get_logger(), "Hello, world! This is a simple ROS 2 node in C++.");
    rclcpp::shutdown();
    return 0;
}